torch/csrc/jit/passes/onnx/deduplicate_initializers.cpp

```diff
@@ -0,0 +1,94 @@
+#include <torch/csrc/jit/jit_log.h>
+#include <torch/csrc/jit/passes/onnx/deduplicate_initializers.h>
+#include <torch/csrc/jit/passes/onnx/helper.h>
+
+#include <c10/util/irange.h>
+
+namespace torch {
+namespace jit {
+
+namespace onnx {
+using namespace ::c10::onnx;
+}
+
+void DeduplicateInitializers(
+    std::shared_ptr<Graph>& g,
+    ValueToParamPairMap& valsToParamsMap,
+    bool (*comp)(at::Tensor&, at::Tensor&)) {
+  auto is_same_tensor_as = [&valsToParamsMap, comp](Value* v1) {
+    return [&valsToParamsMap, v1, comp](Value* v2) {
+      if ((valsToParamsMap.find(v1) == valsToParamsMap.end()) ||
+          (valsToParamsMap.find(v2) == valsToParamsMap.end())) {
+        return false;
```

Comment:
@BowenBao Hi, I was wondering why the last initializer is treated differently and never deduplicated? cc @xenova

Reply:
Hi @fxmarty, the code was sanity check to ensure the values are initializers (can be found in `valsToParamsMap`). Excluding last initializer is not the goal, did you find a model that exhibits such behavior? Is it related to #108342?

Reply:
Yes related to https://github.com/pytorch/pytorch/issues/108342. I think I misread the code, not super familiar with C++.